File: app/Packages/ONTFeatures/Sources/YouFeature/Presentation/YouTab.swift
import ONTDesignSystem
import ONTKit
import ReadingFeature
import SwiftUI

/// **Vous** — le compte, et l'état du corpus qu'on porte.
///
/// L'authentification est posée mais pas branchée : les trois fournisseurs
/// (Apple, Google, GitHub) supposent un backend qui délivre et vérifie les
/// jetons. « Sign in with Apple » est obligatoire dès qu'un autre fournisseur
/// tiers est proposé — c'est une règle de l'App Store, pas une préférence,
/// d'où sa place en premier.
public struct YouTab: View {
    @Environment(YouModel.self) private var model
    @Environment(AccountModel.self) private var account
    @Environment(ReadingModel.self) private var reading

    /// Ce que fait l'app quand le rappel change. Injecté depuis la cible
    /// d'app : `UserNotifications` n'a rien à faire dans une feature.
    private let onDailyChange: (DailyVerseSchedule) async -> Bool
    private let onParutions: (Bool) async -> Bool

    public init(
        onDailyChange: @escaping (DailyVerseSchedule) async -> Bool = { _ in false },
        onParutions: @escaping (Bool) async -> Bool = { _ in false }
    ) {
        self.onDailyChange = onDailyChange
        self.onParutions = onParutions
    }

    /// Ce qui est **montrable** : les pierres tombales n'en sont pas.
    private var nombreDeSurlignages: Int {
        reading.surlignagesParLivre().reduce(0) { $0 + $1.surlignages.count }
    }

    public var body: some View {
        @Bindable var reading = reading

        return NavigationStack {
            List {
                AccountSection()

                // **Ce qui interrompt le lecteur est rangé à part.**
                //
                // Les deux réglages ci-dessous ne servaient qu'à un seul
                // écran, « Verset du jour », rangé sous « Lecture ». Accepter
                // d'envoyer un identifiant d'appareil à un serveur se faisait
                // donc au troisième niveau d'une section qui parle de
                // typographie. Le verset du jour y était mal rangé aussi : il
                // interrompt, il ne se lit pas.
                Section(header: Text("Notifications").font(ONTUI.enteteDeListe)) {
                    NavigationLink {
                        DailyVerseSettings(onChange: onDailyChange)
                    } label: {
                        Label("Verset du jour", systemImage: "sun.horizon")
                            .ontCarteDeLigne()
                    }
                    NavigationLink {
                        ParutionsSettings(onParutions: onParutions)
                    } label: {
                        Label("Parutions", systemImage: "book.closed")
                            .ontCarteDeLigne()
                    }
                }
                .ontLigneDeCarte()

                Section(header: Text("Lecture").font(ONTUI.enteteDeListe)) {
                    NavigationLink {
                        ReadingSettingsSheet()
                    } label: {
                        Label("Réglages de lecture", systemImage: "textformat.size")
                            .ontCarteDeLigne()
                    }
                    // **Ce que le lecteur a marqué lui appartient**, et c'était
                    // jusqu'ici la seule chose de l'app qu'il ne pouvait pas
                    // revoir : les surlignages n'existaient que là où ils
                    // avaient été posés, un verset à la fois, dans un chapitre
                    // qu'il fallait retrouver de mémoire.
                    // **Ici et non dans la barre de partage.**
                    //
                    // Le moment où l'on veut ces réglages est celui du partage
                    // — mais la barre de lecture porte déjà quatre tuiles, et
                    // une cinquième pour un écran qu'on ouvre trois fois dans
                    // sa vie coûterait la place de celles qu'on touche tous les
                    // jours.
                    NavigationLink {
                        OptionsDePartage()
                    } label: {
                        Label("Options de partage", systemImage: "square.and.arrow.up")
                            .ontCarteDeLigne()
                    }
                    NavigationLink {
                        MesSurlignages()
                    } label: {
                        LabeledContent {
                            Text("\(nombreDeSurlignages)").monospacedDigit()
                                .font(ONTUI.ligneDeListe)
                        } label: {
                            Label("Surlignages", systemImage: "highlighter")
                        }
                        .ontCarteDeLigne()
                    }
                }
                .ontLigneDeCarte()

                // **Le registre ouvre la section du corpus, et ne s'y confond pas.**
                //
                // Il était rangé dans les réglages de lecture, entre la
                // disposition des versets et la taille du texte. C'était le
                // ranger avec la typographie : or il ne change pas la façon
                // dont le texte se présente, il change **ce que les livres
                // sont appelés** — donc le corpus lui-même, tel que le lecteur
                // le rencontre.
                //
                // Sa propre carte, détachée des trois compteurs, parce que ce
                // n'est pas une mesure : c'est le seul réglage de cet écran
                // qui décide de ce qu'on lit plutôt que de son état.
                Section {
                    Toggle(isOn: $reading.preferences.french) {
                        Label("Le français reçu", systemImage: "character.book.closed")
                    }
                    .ontCarteDeLigne()
                } header: {
                    Text("Le Corpus")
                        .font(ONTUI.enteteDeListe)
                } footer: {
                    Text(
                        "Allumé, les livres portent le nom qu'on leur connaît — "
                            + "« Apocalypse », « la Loi », « Chapitre 7 ». Éteint, ils "
                            + "portent ce que leur nom hébreu veut dire : « le machazeh "
                            + "de Yohanan », « la Fondation », « Parashah 7 ».\n\n"
                            + "L'écart entre les deux n'est pas une nuance de traduction. "
                            + "La torah est l'instruction qui vise ; le grec l'a rendue par "
                            + "nomos, le code qui contraint, et le français en a hérité "
                            + "« la Loi ».\n\n"
                            + "Ce réglage est une béquille, et il est allumé pour qu'on "
                            + "puisse marcher avant de savoir. En l'éteignant, des mots "
                            + "apparaissent que vous n'avez peut-être jamais lus — parashah, "
                            + "la division que le scribe hébreu traçait en laissant un blanc, "
                            + "mille ans avant qu'on numérote des chapitres."
                    )
                        .font(ONTUI.piedDeListe)
                }
                .ontLigneDeCarte()

                Section {
                    LabeledContent("Slots rédigés") {
                        Text("\(model.writtenBooks) / \(model.allBooks)")
                            .monospacedDigit()
                            .font(ONTUI.ligneDeListe)
                    }
                    .ontCarteDeLigne()
                    LabeledContent("Versets") {
                        Text("\(totalVerses)").monospacedDigit()
                            .font(ONTUI.ligneDeListe)
                    }
                    .ontCarteDeLigne()
                    LabeledContent("Entrées de lexique") {
                        Text("\(model.glossaryCount)").monospacedDigit()
                            .font(ONTUI.ligneDeListe)
                    }
                    .ontCarteDeLigne()
                } footer: {
                    Text(
                        "La Bible ONT est une restitution en cours. Le corpus s'étend "
                            + "à mesure que les "
                            + LibelleDUnite.noms(french: reading.preferences.french)
                            + " sont verrouillés."
                    )
                        .font(ONTUI.piedDeListe)
                }
                .ontLigneDeCarte()

                #if DEBUG
                Section {
                    NavigationLink {
                        DSCatalog()
                    } label: {
                        Label("Design system", systemImage: "paintpalette")
                            .ontCarteDeLigne()
                    }
                    // L'éditeur n'existe qu'une fois connecté, et une
                    // connexion réelle demande un fournisseur, un compte et un
                    // aller-retour réseau. Cette entrée le rend visible sans
                    // rien simuler : le profil est **local**, il se remplit et
                    // se relit à l'identique.
                    NavigationLink {
                        EditeurDuProfil()
                    } label: {
                        Label("Profil", systemImage: "person.crop.circle")
                            .ontCarteDeLigne()
                    }
                } header: {
                    Text("Développement")
                        .font(ONTUI.enteteDeListe)
                } footer: {
                    Text(
                        "Le catalogue des jetons et composants. Un composant ajouté sans sa "
                            + "ligne de catalogue est un composant qu'on oubliera."
                    )
                        .font(ONTUI.piedDeListe)
                }
                .ontLigneDeCarte()
                #endif

                Section(header: Text("Crédits").font(ONTUI.enteteDeListe)) {
                    // Le nom public, et jamais le nom fonctionnel.
                    //
                    // « Sha'eliel » est interne au vault : c'est le nom sous
                    // lequel l'auteur travaille, pas celui sous lequel il
                    // signe. Cet écran est la seule page de l'app où le crédit
                    // paraît, donc le seul endroit où la confusion se voyait.
                    LabeledContent("Traduction", value: "Gloire Bikouta").ontCarteDeLigne()
                    LabeledContent("Hébreu", value: "Ezra SIL — SIL Open Font License")
                        .ontCarteDeLigne()
                    LabeledContent("Titres", value: "Frank Ruhl Libre — OFL").ontCarteDeLigne()
                }
                .ontLigneDeCarte()
            }
            .ontListeDeCartes()
            .ontScreen()
            .navigationTitle("Vous")
            // Demander à l'ouverture de l'onglet, pas au lancement de l'app :
            // c'est le seul écran qui s'en sert, et un lecteur qui n'y vient
            // jamais n'a pas à payer un appel réseau.
            //
            // `.task` et non `.onAppear` : il s'annule si le lecteur repart
            // avant la réponse, et il ne relance pas à chaque retour.
            .task { await account.negocier() }
        }
        .ontColumn()
    }

    private var totalVerses: Int { model.verses }
}

/// Le compte et la synchronisation.
///
/// L'ordre des fournisseurs n'est pas cosmétique : « Continuer avec Apple »
/// doit figurer en premier dès qu'un autre fournisseur tiers est proposé —
/// c'est une règle de la revue App Store.
private struct AccountSection: View {
    @Environment(AccountModel.self) private var account
    @Environment(\.ontTheme) private var theme
    var spacing = ONTSpacing()

    @State private var confirmingErasure = false

    var body: some View {
        @Bindable var account = account

        switch account.state {
        case .signedOut, .failed:
            Section {
                VStack(spacing: spacing.s) {
                    // **Ce que le serveur sait faire**, et non tout ce
                    // qui existe. Un bouton qui ne peut pas aboutir n'a rien à
                    // faire là : le lecteur ne peut pas savoir que les
                    // identifiants manquent chez nous, et l'échec lui
                    // paraîtrait venir du fournisseur.
                    //
                    // Une offre inconnue — hors ligne, ou serveur d'avant la
                    // route — les rend tous : on ne refuse pas ce qu'on ignore.
                    ForEach(account.capacites.fournisseurs, id: \.self) { provider in
                        Button {
                            Task { await account.signIn(with: provider) }
                        } label: {
                            // La capsule est dessinée ici plutôt que laissée à
                            // `.borderedProminent`. Ce style ne colore que le
                            // **titre** du label ; l'icône, elle, garde la
                            // teinte d'accent du formulaire — qui était le
                            // bordeaux de la capsule. L'icône était donc peinte
                            // de la couleur de son propre fond.
                            //
                            // En la posant à la main, le fond et ce qui se pose
                            // dessus viennent de la même paire de rôles et ne
                            // peuvent plus se confondre.
                            // Le texte et l'icône sont en **or** sur le
                            // bordeaux, et non en encre claire : l'or est la
                            // couleur de marque du projet, et ces trois
                            // capsules sont le seul aplat de marque de l'app.
                            //
                            // `onBrandAccent` et non `gold` : sur les thèmes
                            // sombres la capsule **est** l'or, et demander l'or
                            // dessus donnerait un bouton vide.
                            Label("Continuer avec \(provider.label)", systemImage: icon(provider))
                                .foregroundStyle(ONTColors.onBrandAccent(theme.mode))
                                .frame(maxWidth: .infinity)
                                .padding(.vertical, spacing.m)
                                .background(
                                    Capsule().fill(ONTColors.brandInk(theme.mode))
                                )
                                .ontSurvol(dans: Capsule(), souleve: true)
                                .contentShape(.capsule)
                        }
                        .buttonStyle(.ontPresse)
                    }
                }
                .padding(.vertical, spacing.xs)
            } header: {
                Text("Compte")
                    .font(ONTUI.enteteDeListe)
            } footer: {
                // L'échec **s'ajoute** à l'explication, il ne la remplace pas.
                //
                // Il la remplaçait : une connexion ratée effaçait donc la seule
                // phrase qui dit que le compte est facultatif, et laissait
                // croire l'app cassée. C'est précisément ce qu'a vu un
                // examinateur de l'App Store le 19 août 2026.
                VStack(alignment: .leading, spacing: spacing.xs) {
                    if case .failed(let message) = account.state {
                        // La braise de la gamme, dans sa pastille — le rouge
                        // du système n'a pas voix dans cette maison, et un
                        // échec se lit sans crier.
                        Text(message)
                            .font(ONTUI.piedDeListe)
                            .foregroundStyle(theme.danger)
                            .padding(.horizontal, spacing.m)
                            .padding(.vertical, spacing.s)
                            .background(theme.dangerSurface, in: .rect(cornerRadius: 8))
                    }
                    Text(
                        "La lecture, les surlignages et les notes fonctionnent entièrement "
                            + "sans compte. La connexion ne sert qu'à les retrouver sur un "
                            + "autre appareil."
                    )
                        .font(ONTUI.piedDeListe)
                }
            }
            .ontLigneDeCarte()

        case .working:
            Section(header: Text("Compte").font(ONTUI.enteteDeListe)) {
                HStack(spacing: spacing.m) {
                    ProgressView()
                    Text("Connexion…").foregroundStyle(.secondary)
                        .font(ONTUI.ligneDeListe)
                }
            }
            .ontLigneDeCarte()

        case .signedIn:
            // **Le profil ouvre le compte, avant la synchronisation.**
            //
            // C'est ce que le lecteur vient voir : « je suis connecté, et
            // voilà qui je suis ». Le consentement et les échanges sont de
            // l'administration — vraie, nécessaire, mais qui ne répond pas à
            // la question qu'on se pose en ouvrant l'écran.
            //
            // **Et seulement connecté.** Un profil vide en tête d'un écran
            // sans compte ne dit rien, là où les trois boutons de connexion
            // disent quoi faire. L'auteur a d'ailleurs pris l'aperçu de
            // développement pour le vrai, ce qui est le signe qu'un profil
            // hors compte n'a pas de sens à cet endroit.
            Section {
                EnTeteDuProfil()
                    .ontCarteDeLigne()
            }
            .ontLigneDeCarte()

            Section {
                // Le consentement est explicite et séparé : les annotations
                // d'un lecteur de Bible révèlent des convictions religieuses
                // (RGPD, article 9), et ne peuvent pas partir sur la foi
                // d'une case noyée dans des conditions générales.
                Toggle("Synchroniser mes annotations", isOn: $account.consent)
                    .ontCarteDeLigne()

                if account.consent {
                    Button {
                        Task { await account.synchronise() }
                    } label: {
                        LabeledContent("Synchroniser maintenant") {
                            if account.syncing {
                                ProgressView()
                            } else if let last = account.lastSync {
                                Text(last.formatted(date: .omitted, time: .shortened))
                                    .foregroundStyle(.secondary)
                                    .font(ONTUI.ligneDeListe)
                            }
                        }
                    }
                    .disabled(account.syncing)
                }
            } header: {
                Text("Compte")
                    .font(ONTUI.enteteDeListe)
            } footer: {
                Text(
                    "Vos surlignages et vos notes disent ce que vous lisez et ce qui vous "
                        + "arrête. Tant que cet interrupteur est éteint, ils ne quittent pas "
                        + "cet appareil."
                )
                    .font(ONTUI.piedDeListe)
            }
            .ontLigneDeCarte()

            Section {
                Button("Se déconnecter") { account.signOut() }
                    .ontCarteDeLigne()
                // La braise et non le rouge du système — même refus, même voix.
                Button("Supprimer mon compte", role: .destructive) {
                    confirmingErasure = true
                }
                .tint(theme.danger)
                .ontCarteDeLigne()
            } footer: {
                Text(
                    "La suppression efface la copie sur le serveur. Vos annotations restent "
                        + "sur cet appareil."
                )
                    .font(ONTUI.piedDeListe)
            }
            .ontLigneDeCarte()
            .confirmDeletion($confirmingErasure) {
                Task { await account.eraseAccount() }
            }
        }
    }

    private func icon(_ provider: AuthProvider) -> String {
        switch provider {
        case .apple: "apple.logo"
        case .google: "g.circle.fill"
        case .github: "chevron.left.forwardslash.chevron.right"
        }
    }
}

extension View {
    fileprivate func confirmDeletion(
        _ presented: Binding<Bool>,
        action: @escaping () -> Void
    ) -> some View {
        confirmationDialog(
            "Supprimer le compte ?",
            isPresented: presented,
            titleVisibility: .visible
        ) {
            Button("Supprimer", role: .destructive, action: action)
            Button("Annuler", role: .cancel) {}
        } message: {
            Text(
                "La copie de vos annotations sur le serveur sera effacée définitivement, "
                    + "ainsi que votre profil sur cet appareil — photo, nom et bio. "
                    + "Vos surlignages, eux, restent sur l'appareil."
            )
                .font(ONTUI.ligneDeListe)
        }
    }
}

File: android/ontfeatures/src/main/kotlin/com/labibleont/ont/features/you/YouTab.kt
package com.labibleont.ont.features.you

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
import androidx.compose.material.icons.automirrored.filled.MenuBook
import androidx.compose.material.icons.filled.FormatSize
import androidx.compose.material.icons.filled.Palette
import androidx.compose.material.icons.filled.WbTwilight
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.labibleont.ont.designsystem.metrics.ONTRadius
import com.labibleont.ont.designsystem.metrics.ontSpacing
import com.labibleont.ont.designsystem.surfaces.ONTGroup
import com.labibleont.ont.designsystem.surfaces.ONTGroupDivider
import com.labibleont.ont.designsystem.surfaces.ONTLargeTitle
import com.labibleont.ont.designsystem.surfaces.ONTPage
import com.labibleont.ont.designsystem.surfaces.ONTRow
import com.labibleont.ont.designsystem.surfaces.ONTSectionHeader
import com.labibleont.ont.designsystem.theme.LocalReadingTheme
import com.labibleont.ont.designsystem.tokens.ONTColors
import com.labibleont.ont.kit.corpus.LibelleDUnite
import com.labibleont.ont.kit.reader.ReadingPreferences

/** Ce vers quoi une ligne de l'onglet Vous conduit. */
public enum class DestinationVous {
    LECTURE,
    VERSET_DU_JOUR,
    PARUTIONS,

    /** Le catalogue du design system — en développement seulement. */
    CATALOGUE,
}

/**
 * L'onglet Vous — un **hall**, pas un déversoir de réglages.
 *
 * ## Pourquoi des sous-écrans et non une page unique
 *
 * La première version d'Android empilait tout : niveaux, taille, thème, fonte,
 * rappel. C'était lisible et ça ne ressemblait à rien — une colonne
 * d'interrupteurs séparés par des traits pleine largeur.
 *
 * iOS en fait un sommaire : quelques groupes encartés, chaque ligne menant à un
 * écran qui ne traite qu'une chose. Ce n'est pas une préférence esthétique.
 * C'est ce qui permet à l'écran du **verset du jour** d'exister séparément de
 * celui des **parutions** — la distinction que la PR #74 a établie côté iOS, et
 * qu'une page unique aurait effacée.
 *
 * On reprend donc la structure. Ce qui change, ce sont les gestes : le retour
 * système ferme un sous-écran, et les lignes suivent la cible tactile
 * d'Android.
 */
@Composable
public fun YouTab(
    preferences: ReadingPreferences,
    slotsRediges: Int,
    slotsTotal: Int,
    versets: Int,
    /**
     * Le nombre d'entrées du lexique.
     *
     * Il vaut zéro tant que le lexique n'a pas été chargé — c'est pourquoi
     * l'écran des onglets le charge aussi en arrivant ici, et pas seulement en
     * ouvrant l'onglet Lexique. Un compteur qui affiche zéro parce que la
     * donnée n'est pas venue ne se distingue pas d'un corpus vide.
     */
    entreesDeLexique: Int,
    /**
     * Le fournisseur de la session ouverte, ou `null` si personne ne l'est.
     *
     * Une chaîne et non un type du domaine : cet écran n'a pas à connaître
     * l'énumération des fournisseurs pour afficher un mot.
     */
    fournisseurConnecte: String?,
    onAller: (DestinationVous) -> Unit,
    onConnecter: (String) -> Unit,
    onDeconnecter: () -> Unit,
    onPasEncore: () -> Unit,
    /**
     * Vrai en build de développement.
     *
     * Le catalogue n'a rien à faire dans une app livrée : il ne sert qu'à
     * repérer une dérive de style avant qu'elle n'atteigne un écran de lecture.
     */
    enDeveloppement: Boolean = false,
    modifier: Modifier = Modifier,
) {
    val theme = LocalReadingTheme.current
    val espace = ontSpacing

    Column(
        modifier = modifier
            .fillMaxWidth()
            .verticalScroll(rememberScrollState()),
    ) {
        ONTPage {
            ONTLargeTitle("Vous")

            ONTSectionHeader("Compte")
            ONTGroup {
                Column(
                    modifier = Modifier.padding(espace.m),
                    verticalArrangement = Arrangement.spacedBy(espace.s),
                ) {
                    if (fournisseurConnecte == null) {
                        BoutonDeConnexion("Continuer avec Google") { onConnecter("google") }
                        BoutonDeConnexion("Continuer avec GitHub") { onConnecter("github") }
                    } else {
                        // ## Ce qu'on montre d'un compte ouvert
                        //
                        // Le fournisseur, et rien d'autre. Le backend ne rend
                        // que des jetons à la connexion — pas d'adresse — et
                        // aller la chercher demanderait une route de plus pour
                        // afficher une ligne. iOS la montre parce qu'Apple la
                        // lui donne au passage.
                        Text(
                            "Connecté avec " + when (fournisseurConnecte) {
                                "google" -> "Google"
                                "github" -> "GitHub"
                                else -> fournisseurConnecte
                            },
                            color = ONTColors.ink(theme),
                            fontSize = 15.sp,
                        )
                        BoutonDeConnexion("Se déconnecter") { onDeconnecter() }
                    }
                }
            }
            Text(
                // Le dire avant qu'on se demande : rien n'oblige à créer un
                // compte. C'est une propriété de l'app, pas une concession.
                if (fournisseurConnecte == null) {
                    "La lecture, les surlignages et les notes fonctionnent " +
                        "entièrement sans compte. La connexion ne sert qu'à les " +
                        "retrouver sur un autre appareil."
                } else {
                    "Vos marques restent sur cet appareil. Se déconnecter ne les " +
                        "efface pas."
                },
                color = ONTColors.inkSoft(theme),
                fontSize = 13.sp,
                modifier = Modifier.padding(
                    start = espace.l,
                    end = espace.l,
                    top = espace.s,
                ),
            )

            Spacer(Modifier.height(espace.xl))

            ONTSectionHeader("Notifications")
            ONTGroup {
                ONTRow(
                    titre = "Verset du jour",
                    detail = if (preferences.daily.enabled) {
                        "à %02d:%02d".format(preferences.daily.hour, preferences.daily.minute)
                    } else {
                        "éteint"
                    },
                    icone = Icons.Filled.WbTwilight,
                    onClick = { onAller(DestinationVous.VERSET_DU_JOUR) },
                    fin = { Chevron() },
                )
                ONTGroupDivider()
                ONTRow(
                    titre = "Parutions",
                    detail = "quand un livre paraît",
                    icone = Icons.AutoMirrored.Filled.MenuBook,
                    onClick = { onAller(DestinationVous.PARUTIONS) },
                    fin = { Chevron() },
                )
            }

            Spacer(Modifier.height(espace.xl))

            ONTSectionHeader("Lecture")
            ONTGroup {
                ONTRow(
                    titre = "Réglages de lecture",
                    detail = "niveaux, taille, thème, fonte",
                    icone = Icons.Filled.FormatSize,
                    onClick = { onAller(DestinationVous.LECTURE) },
                    fin = { Chevron() },
                )
            }

            Spacer(Modifier.height(espace.xl))

            ONTSectionHeader("Le corpus")
            ONTGroup {
                ONTRow(titre = "Slots rédigés", fin = { Valeur("$slotsRediges / $slotsTotal") })
                ONTGroupDivider()
                ONTRow(titre = "Versets", fin = { Valeur("$versets") })
                ONTGroupDivider()
                ONTRow(titre = "Entrées de lexique", fin = { Valeur("$entreesDeLexique") })
            }

            // ## Le pied qui dit pourquoi les chiffres sont bas
            //
            // Sans lui, « 4 / 70 » se lit comme un manque. Avec lui, comme un
            // chantier — ce qu'il est. iOS le porte depuis toujours ; Android
            // affichait les mêmes chiffres nus.
            //
            // Et le mot suit le registre : « chapitres » ou « parashiot ». Le
            // pluriel de *parashah* prend la marque hébraïque, pas le `s`
            // français — franciser l'intraduisible ici défferait exactement ce
            // que le réglage vient de faire.
            Text(
                "La Bible ONT est une restitution en cours. Le corpus s'étend " +
                    "à mesure que les ${LibelleDUnite.noms(preferences.french)} " +
                    "sont verrouillés.",
                color = ONTColors.inkSoft(theme),
                fontSize = 13.sp,
                modifier = Modifier.padding(
                    start = espace.l,
                    end = espace.l,
                    top = espace.s,
                ),
            )

            // ## Les crédits, que l'app taisait
            //
            // C'est le seul écran où le crédit paraît, donc le seul endroit où
            // la confusion des noms se verrait : **Gloire Bikouta** en public,
            // jamais « Sha'eliel », qui est le nom interne au vault.
            //
            // Et les deux licences de fontes ne sont pas une politesse : Ezra
            // SIL et Frank Ruhl Libre sont sous OFL, qui **exige** que la
            // mention accompagne la distribution. Une app qui les embarque sans
            // les créditer est en défaut, et le défaut voyage avec chaque
            // installation.
            Spacer(Modifier.height(espace.xl))
            ONTSectionHeader("Crédits")
            ONTGroup {
                ONTRow(titre = "Traduction", fin = { Valeur("Gloire Bikouta") })
                ONTGroupDivider()
                ONTRow(titre = "Hébreu", fin = { Valeur("Ezra SIL — SIL Open Font License") })
                ONTGroupDivider()
                ONTRow(titre = "Titres", fin = { Valeur("Frank Ruhl Libre — OFL") })
            }

            if (enDeveloppement) {
                Spacer(Modifier.height(espace.xl))
                ONTSectionHeader("Développement")
                ONTGroup {
                    ONTRow(
                        titre = "Design system",
                        detail = "jetons, surfaces, fontes, rendu",
                        icone = Icons.Filled.Palette,
                        onClick = { onAller(DestinationVous.CATALOGUE) },
                        fin = { Chevron() },
                    )
                }
            }

            Spacer(Modifier.height(espace.xxl))
        }
    }
}

@Composable
private fun Chevron() {
    val theme = LocalReadingTheme.current
    Icon(
        Icons.AutoMirrored.Filled.KeyboardArrowRight,
        contentDescription = null,
        tint = ONTColors.inkSoft(theme),
    )
}

@Composable
private fun Valeur(texte: String) {
    val theme = LocalReadingTheme.current
    Text(texte, color = ONTColors.inkSoft(theme), fontSize = 15.sp)
}

/**
 * Un bouton de connexion.
 *
 * Éteint tant que le compte n'existe pas, et **il le dit** : un bouton qui ne
 * répond pas sans expliquer pourquoi se lit comme un défaut de l'app.
 *
 * Pas de « Continuer avec Apple » : sur Android, il faudrait passer par le web,
 * et le lecteur qui a un compte Apple l'aura créé sur son iPhone. On l'ajoutera
 * quand la synchronisation existera, pas avant — annoncer trois fournisseurs
 * dont un ne marche pas est pire que d'en annoncer deux.
 */
@Composable
private fun BoutonDeConnexion(intitule: String, onAppui: () -> Unit) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .heightIn(min = 48.dp)
            .clip(RoundedCornerShape(ONTRadius.pill))
            // **Plein**, pas délavé. Un bouton grisé se lit comme un défaut de
            // l'app ; celui-ci répond, et ce qu'il répond est qu'il n'est pas
            // encore branché. Dire « pas encore » vaut mieux que ne rien dire.
            .background(ONTColors.burgundy)
            .clickable(onClick = onAppui)
            .padding(horizontal = 16.dp, vertical = 12.dp),
        horizontalArrangement = Arrangement.Center,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Text(
            intitule,
            color = ONTColors.gold,
            fontSize = 16.sp,
            textAlign = TextAlign.Center,
        )
    }
}
